### ISSUE TITLE
Building Symphony-5.1.3 with MSVC v6: coinindexedvector.cpp

### ISSUE THREAD

**Issue by svigerske** [coin-or/SYMPHONY#63]
Issue created by migration from Trac.

Original creator: pchan

Original creation time: 2007-02-09 20:33:46

Assignee: menal

Version: 5.1.2

Hi,
I've downloaded the 5.1.3 tarball, expanded it and opened "symphony.dsw"
within MSVC6. 7 errors (all C2632: 'long' followed by 'long' is illegal)
in the following locations:

..\symphony-5.1.3\coinutils\src\coinindexedvector.cpp(543) : error C2632
....\coinindexedvector.cpp(543) : error C2632
....\coinindexedvector.cpp(544) : error C2632
....\coinindexedvector.cpp(1388) : error C2632
....\coinindexedvector.cpp(1388) : error C2632
....\coinindexedvector.cpp(1389) : error C2632
....\coinindexedvector.cpp(1393) : error C2632

Its got to do with
   CoinInt64 xx = reinterpret_cast<CoinInt64>(temp);

Thank you!

**Comment by svigerske** [comment]
Comment by pchan created at 2007-02-10 10:10:44

The same error arises from symphony-5.1.0, in the same coinIndexedVector.cpp file.

With the binary download, putting the same Debug libCoinUtils.lib in the output
directory was no help, the make file rebuilds each object *.obj before building
the library. Inspite the fact that downloaded lib was dated well after the modified
dates of the source files. Very bizzare indeed !

**Comment by svigerske** [comment]
Comment by menal created at 2007-02-10 15:17:29

Pchan, 

The problem is that MSVC-v6 doesn't support the 'long long' type. Possible fixes: 
- Change the definition of COIN_INT64_T to long in BuildTools/headers/configall_system_msc.h. I am not sure if this may cause a problem later.
- You can download and use MSVC v8 from 'http://msdn.microsoft.com/vstudio/express/visualc/download'. It is free! 
- File a ticket to CoinUtils project (https://projects.coin-or.org/CoinUtils) and wait somebody to fix it. 

Menal.

**Comment by svigerske** [comment]
Comment by pchan created at 2007-02-10 21:03:56

Thanks for point out the problem and where to fix it !
I used __int64 as suggested by the MSDN page (2001 library)
"INFO: Understanding and Using GetDiskFreeSpace and GetDiskFreeSpaceEx"
in the last paragraph:
"Notes on 64-bit Integer Math".
I'll post back if it's working (I will try SPP+Cuts)!

I'm not moving to v8 as yet, this kind of move depends on my company's 
development and support strategies. Sorry.

**Comment by svigerske** [comment]
Comment by pchan created at 2007-02-10 21:32:26

To be precise, I keyed in the type int64 with two preceeding underscores in the above comment. TRAC translated it in it's fashion. 
If you want the updated file, let me know!

No more compilation errors, in any of the packages including SPP+Cut. 
With the "sample.spp", I got the optimal cost 6314.000000. 
1) Is this the expected value?
2) Are there any other examples to further test it, with known optimal values?
3) I plan to move from Set Partitioning to Set Covering. Any suggestions are welcome!

Thank you again for your help!
